test.log

```diff
@@ -0,0 +1,735 @@
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestCreateDirWhichIsFile/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestCreateDirTwice/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestCreateDirPathIsInvalid/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestRecursivelyCreateDir/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestDeleteFileWhichIsDirectory/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestDeleteFilePathIsInvalid/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestDeleteRecursivelyEmpty/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestDeleteRecursivelyNotEmpty/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestDeleteRecursivelyPathIsInvalid/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestRenameFileDestinationParentNotFound/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestRenameFileSourceIsDirectory/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestRenameFileTargetIsDirectory/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestRenameFileSourcePathIsInvalid/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestRenameFileTargetPathIsInvalid/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestCopyFileSourceIsDirectory/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestCopyFileTargetIsDirectory/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestCopyFileSourcePathIsInvalid/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestCopyFileTargetPathIsInvalid/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestFileExistsButIsDirectory/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestFileExistsPathIsInvalid/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestFilesExistAllFailureModes/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestStatDirectory/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestStatPathIsInvalid/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestIsDirectory/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestIsDirectoryPathIsInvalid/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestGetFileSizeDirectory/0, where GetParam() = "gs"
+[  FAILED  ] ModularFileSystem/ModularFileSystemTest.TestGetFileSizePathIsInvalid/0, where GetParam() = "gs"
```

Comment:
Actually, more tests would be failed if I did not force `Sync()` on every `WritableFile`, there are 2 things to mentions:
- `WritableFile` will write everything to a temporary local file and only upload to server when `Sync()`, `Close()` or `Flush()` called. When you call `NewWritableFile`, local filesystem will create that file immediately, but `gcs` will create a temporary file. Example, if we don't call `Sync()`, this test will failed because `new_file` isn't uploaded to the server. So `NewReadOnlyMemoryRegionFromFile` will return Not Found instead of reading an empty file
https://github.com/tensorflow/tensorflow/blob/60ffd3d4d6c27aa43c5ca6893e1a29ed654d1334/tensorflow/c/experimental/filesystem/modular_filesystem_test.cc#L382-L392

- `RandomAccessFile` will only be load when `Read()`. `NewRandomAccessFile` just create a `RandomAccessFile` with a bucket, an object and a `gcs::Client`. When `Read()` is call, `gcs::Client` will try to read that object in cloud and see if it is existed or not. Example, this test below, the file isn't existed but `NewRandomAccessFile` does not check it.
https://github.com/tensorflow/tensorflow/blob/62d50aa8e28209c701e8919d315ed0cb3ca804de/tensorflow/c/experimental/filesystem/modular_filesystem_test.cc#L309-L314

Reply:
That makes sense. Adding another item for refining the design.